napari/layers/_pyramid_layer/model.py

```diff
@@ -0,0 +1,249 @@
+
+    @property
+    def pyramid(self):
+        """list: List of np.ndarry image data, with base of pyramid at `0`.
+        """
+        return self._pyramid
+
+    @pyramid.setter
+    def pyramid(self, pyramid):
+        self._pyramid = pyramid
+
+        self.refresh()
+
+    @property
+    def pyramid_level(self):
+        """int: Level of pyramid to show, with base of pyramid at `0`.
+        """
+        return self._pyramid_level
+
+    @pyramid_level.setter
+    def pyramid_level(self, level):
+        if self._pyramid_level == level:
+            return
+        self._pyramid_level = level
+        self.scale = [self._image_downsamples[self.pyramid_level]] * 2
+        self._top_left = self.find_top_left()
+        self._update_image_from_pyramid()
+        self._update_coordinates()
+        self.refresh()
+
+    @property
+    def top_left(self):
+        """int: Location of top left pixel.
+        """
+        return self._top_left
+
+    @top_left.setter
+    def top_left(self, top_left):
+        if np.all(self._top_left == top_left):
+            return
+        self._top_left = top_left
+        self._update_image_from_pyramid()
+        self._update_coordinates()
+        self.refresh()
+
+    def _update_image_from_pyramid(self):
+        """Updates the image data from the pyramid based on requested tile
+        and pyramid level
+        """
+        if np.any(self._image_shapes[self.pyramid_level] >
+                  self._max_tile_shape):
+            slices = [slice(self._top_left[i], self._top_left[i] +
+                            self._max_tile_shape[i], 1) for i in range(2)]
+            self._image = self.pyramid[self.pyramid_level][tuple(slices)]
+            self.translate = self._top_left[::-1]*self.scale[:2]
+        else:
+            self._image = self.pyramid[self.pyramid_level]
+            self.translate = [0, 0]
+
+    def _get_shape(self):
+        """Shape of base of pyramid
+
+        Returns
+        ----------
+        shape : np.ndarry
+            Shape of base of pyramid
+        """
+        if self.multichannel:
+            return self.pyramid[0].shape[:-1]
+        return self.pyramid[0].shape
+
+    def get_value(self):
+        """Returns coordinates, values, and a string for a given mouse position
+        and set of indices.
+
+        Returns
+        ----------
+        coord : sequence of int
+            Position of mouse cursor in data.
+        value : int or float or sequence of int or float
+            Value of the data at the coord.
+        msg : string
+            String containing a message that can be used as
+            a status update.
+        """
+        coord = np.round(self.coordinates).astype(int)
+        if self.multichannel:
+            shape = self._image_view.shape[:-1]
+        else:
+            shape = self._image_view.shape
+        coord[-2:] = np.clip(coord[-2:], 0, np.asarray(shape) - 1)
+        value = self._image_view[tuple(coord[-2:])]
```

Comment:
is it actually necessary to convert `coord` to a tuple?

Reply:
yes, i get errors if I don't, something about how it's actually doing the slicing seems different

Reply:
Yep, `image[[2, 5]]` means (*) "the second and fifth row of `image`, while `image[(2, 5)]` is the same as `image[2, 5]`, ie the element at coordinate 2, 5.

(*): in a future version of NumPy, for consistency with what happens in the current version with `image[array([2, 5])]`.

Reply:
huh, I guess that makes sense thinking back to how wonky argument handling is with regards to `__getitem__`:
```python
In [1]: class Foo:
   ...:     def __getitem__(self, key):
   ...:         return key
   ...:

In [2]: f = Foo()

In [3]: f[1, 2]
Out[3]: (1, 2)

In [4]: f[(1, 2)]
Out[4]: (1, 2)
```